src/basic/cgroup-util.c

```diff
@@ -211,6 +211,17 @@ int cg_read_subgroup(DIR *d, char **fn) {
                 if (dot_or_dot_dot(de->d_name))
                         continue;
 
+                if (ignore_net_cls) {
+                        _cleanup_free_ char *p = NULL;
+
+                        p = path_join(de->d_name, "net_cls.classid");
```

Comment:
This doesn't look right. The bug is about people mount cgroup-v1 controllers in a directory below the cgroup-v2 in a hierarchy. In particular, any v1 controller will behave like this, not just net_cls.

This can never work — as described in the cgroup delegation documents, systemd expects to have exclusive ownership of the hierarchy (except where delegation is enabled). I think we should close this as CANTFIX.

Reply:
Ah, I did not know that net_cls is a cgroup v1 controller... Thanks.